src/config.js

```diff
@@ -0,0 +1,201 @@
+		 *         toolbar: {
+		 *             collapsed: true
+		 *         }
+		 *     } );
+		 *
+		 * Passing an object as the value will amend the configuration, not replace it.
+		 *
+		 *     config.set( 'toolbar', {
+		 *         collapsed: true,
+		 *     } );
+		 *
+		 *     config.set( 'toolbar', {
+		 *         color: 'red',
+		 *     } );
+		 *
+		 *     config.toolbar.collapsed; // true
+		 *     config.toolbar.color; // 'red'
+		 *
+		 * @param {String|Object} nameOrConfigurations The configuration name or an object from which take properties as
+		 * configuration entries.
+		 * @param {*} [value] The configuration value. Used if a name is passed to nameOrConfigurations. If undefined,
+		 * the configuration is set to `null`.
+		 */
+		set: function( name, value ) {
+			// Just pass the call to the original set() in case of an object. It'll deal with recursing through the
+			// object and calling set( name, value ) again for each property.
+			if ( utils.isObject( name ) ) {
+				Model.prototype.set.apply( this, arguments );
+
+				return;
+			}
+
+			// The target for this configuration is, for now, this object.
+			//jscs:disable safeContextKeyword
+			var target = this;
+			//jscs:enable
+
+			// The configuration name should be split into parts if it has dots. E.g: `resize.width`.
+			var parts = name.toLowerCase().split( '.' );
+
+			// Take the name of the configuration out of the parts. E.g. `resize.width` -> `width`
+			name = parts.pop();
+
+			// Retrieves the final target for this configuration recursively.
+			for ( var i = 0; i < parts.length; i++ ) {
+				// The target will always be an instance of Config.
+				if ( !( target[ parts[ i ] ] instanceof Config ) ) {
+					target.set( parts[ i ], new Config() );
+				}
+
+				target = target[ parts[ i ] ];
+			}
+
+			// Values set as pure objects will be treated as Config subsets.
+			if ( utils.isPlainObject( value ) ) {
+				// If the target is an instance of Config (a deep config subset).
+				if ( target[ name ] instanceof Config ) {
+					// Amend the target with the value, instead of replacing it.
+					target[ name ].set( value );
+
+					return;
+				}
+
+				value = new Config( value );
+			}
+
+			// Values will never be undefined.
+			if ( typeof value == 'undefined' ) {
+				value = null;
+			}
+
+			// Call the original set() on the target.
+			Model.prototype.set.call( target, name, value );
+		},
+
+		/**
+		 * Gets the value for a configuration entry.
+		 *
+		 *     config.get( 'name' );
+		 *
+		 * Deep configurations can be retrieved by separating each part with a dot.
+		 *
+		 *     config.get( 'toolbar.collapsed' );
+		 *
+		 * @param {String} name The configuration name.
+		 * @returns {*} The configuration value or `undefined` if the configuration entry was not found.
+		 */
+		get: function( name ) {
+			// The target for this configuration is, for now, this object.
+			//jscs:disable safeContextKeyword
+			var source = this;
```

Comment:
`that` is a trick to reference `this` inside inner-functions. That's not the case here. 

You should read the rest of the code... here `source` (on `get()`) and `target` (on `set()`) refer to the source/target object for the configurations, which at start is  `this` but then eventually moves to child properties of `this`. Therefore a more semantical name is used for the variable and unfortunately the jscs rules we have applied are too strict for this.